### PR TITLE
multipass(MultipassProvider): support kinetic, lunar, and devel images

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

1. Add support for `kinetic`, `lunar`, and `devel` for Multipass with a new `RemoteImage` class.
2. Check if Multipass image are stable and use the `allow_unstable` parameter.

Split into 4 commits for your reviewing pleasure!

(CRAFT-1637)

### CODE REVIEW

File: tests/unit/multipass/test_multipass_provider.py [hunk 1]
```diff
@@ -110,36 +126,31 @@ def test_create_environment(mocker):
...
+    """Verify `launched_environment()` function."""
     provider = MultipassProvider()
-
     with provider.launched_environment(
         project_name="test-project",
         project_path=tmp_path,
         base_configuration=mock_buildd_base_configuration,
         build_base=build_base,
         instance_name="test-instance-name",
+        allow_unstable=not remote_image.is_stable,
```
Comment (lengau) [inline tests/unit/multipass/test_multipass_provider.py:146]: This could just be `True`, right?

**Review by lengau — APPROVED** [review]
NICE! Let's go!

**Comment by mr-cal** [comment]
Thanks for the review!

Test failure looks like a mistake on my part in the test itself.  Will tweak it tomorrow.

File: craft_providers/multipass/__init__.py [hunk 2]
```diff
@@ -23,18 +23,14 @@
 from .installer import install, is_installed  # noqa: F401
 from .multipass import Multipass  # noqa: F401
 from .multipass_instance import MultipassInstance  # noqa: F401
-from .multipass_provider import (  # noqa: F401
-    PROVIDER_BASE_TO_MULTIPASS_BASE,
-    MultipassProvider,
-)
+from .multipass_provider import MultipassProvider  # noqa: F401
```
[the remark this answers is not in the document]
Comment (sergiusens) [reply craft_providers/multipass/__init__.py:26]: this probably means this is missing the __all__ entry

[the remark this answers is not in the document]
Comment (sergiusens) [reply craft_providers/multipass/__init__.py:26]: you should actually be able to remove all the F401s from this PR as `__all__` is actually there already

[the remark this answers is not in the document]
Comment (mr-cal) [reply craft_providers/multipass/__init__.py:26]: Thanks for the pointers, I removed all F401s here: https://github.com/canonical/craft-providers/pull/243

[the remark this answers is not in the document]
Comment (mr-cal) [reply craft_providers/multipass/__init__.py]: I merged https://github.com/canonical/craft-providers/pull/243 and rebased this PR on top of it.

File: tests/integration/multipass/test_launch.py [hunk 3]
```diff
@@ -57,21 +57,14 @@ def core20_instance(instance_name):
             instance.delete()
 
 
-@pytest.mark.parametrize(
```
Comment (mr-cal) [inline tests/integration/multipass/test_launch.py:60]: Similar to what I did for LXD, I moved this parametrization to the `test_multipass_provider.py` module

**Comment by mr-cal** [comment]
Well, this is unfortunate.  It looks like some Multipass images (`snapcraft:devel`, `daily:devel`, and `daily:lunar`) are not available on MacOS, which is where the Multipass integration tests are run.

I reached out to the Multipass team for advice.

**Comment by mr-cal** [comment]
Multipass team confirmed that devel images are not available on MacOS and Windows, for now.

Per Sergio's advice, I disabled launching Multipass devel images on MacOS.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**